supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/PipelinesApiAsyncBackendJobExecutionActor.scala

```diff
@@ -174,19 +175,38 @@ class PipelinesApiAsyncBackendJobExecutionActor(override val standardParams: Sta
     )
   }
 
-  protected def callInputFiles: Map[FullyQualifiedName, Seq[WomFile]] = jobDescriptor.fullyQualifiedInputs map {
-    case (key, womFile) =>
-      val arrays: Seq[WomArray] = womFile collectAsSeq {
-        case womFile: WomFile =>
-          val files: List[WomSingleFile] = DirectoryFunctions
-            .listWomSingleFiles(womFile, pipelinesApiCallPaths.workflowPaths)
-            .toTry(s"Error getting single files for $womFile").get
-          WomArray(WomArrayType(WomSingleFileType), files)
-      }
+  override lazy val inputsToNotLocalize: Set[WomFile] = {
+    val result = jobDescriptor.findInputFilesByParameterMeta {
+      // TODO: Is "nio" the right name?
```

Comment:
I feel like there potentially could be use cases where one does not want a file to be localized for other reasons than using NIO ? It also somehow conveys the feeling that this file _will_ be accessed by the task using NIO which there's really no way to be sure of from the WDL/engine side of things.
OTOH I don't really care.. :smile:

Reply:
maybe `{ localize: false }`?

Reply:
I totally agree and `nio` is the shortest placeholder I could come up with as I was writing it... what I really want it to convey is that localization is optional, and the engine is *allowed* but not *required* to not localize it

Reply:
isn't it a bit stronger than that? like ideally the engine should not localize the file if possible?

Reply:
`{ localize: preferably_not }`

Reply:
Decidedly not!

Since the stronger `FileURL` was rejected as a WDL feature, this is a possible engine optimization which the task-writer should *not* assume will always happen (but they *are* saying that they'd be cool with it, if it did).

Reply:
@Horneth To be clear there should be no fuzziness on the WDL/engine side of things as this is 100% a Cromwell feature and not a WDL feature.

Reply:
hmm not sure what you mean. It's a Cromwell feature that can only be taken advantage of through WDL for now (once this gets merged). It also involves a WDL syntax

Reply:
Right but there's 0% expectation that it'll work anywhere but Cromwell. It's effectively piggybacking on existing WDL syntax to create a hint

Reply:
And in theory if we adopted a CWL `ParameterMetaHint` or `LocalizationOptionalHint`, we could allow this optimization in CWL workflows too

Reply:
`{ optionalLocalization: true }`? (getting a bit long)